pmlc/target/x86/pipeline.cc

```diff
@@ -170,12 +172,13 @@ void pipelineBuilder(OpPassManager &pm) {
   pm.addPass(pxa::createAffineNormalizePass(/*promote=*/false));
   pm.addPass(createCanonicalizerPass());
 
-  // TODO: Figure out a better way to prevent 'overthreading'
-  auto maxThreads = std::thread::hardware_concurrency();
-  if (maxThreads > 8) {
-    maxThreads = 8;
-  }
-  std::min(std::thread::hardware_concurrency(), maxThreads);
+  // Use OMP thread count
+  auto maxThreads = omp_get_max_theads();
+  // TODO: Get the physical core count (coming in the next commit)
+  // TODO: The hyperthreading is very harmful on perf, so exclude the
+  //       hyper threads.
+  maxThreads = maxThreads / 2;
+  std::max(1, maxThreads);
```

Comment:
The std::max here seems like a no-op

Reply:
Well, it takes care of the case of one core with no hyper-threading. Not that I exopect it to happen a lot, but in benchmarking with one core, it could be an issue.

Reply:
This code is literally dead; it doesn't have any side effects and the return value is unused, therefore, this is a no-op.

Reply:
My bad. Just mechanically modified a line that had the exact problem.